docs/faq.rst

```diff
@@ -93,11 +93,74 @@ The TEDICA step may fail to converge if TEDPCA is either too strict
+[tedana] What is the right number of ICA components & what options let me get it?
+*********************************************************************************
+
+Part of the PCA step in ``tedana`` processing involves identifying the number of
+components that contain meaningful signal.
+The PCA components are then used to calculate the same number of ICA components.
+The ``--tedpca`` option includes several options to identify the "correct" number
+of PCA components.
+``kundu`` and ``kundu-stabilize`` use several echo-based criteria to exclude PCA
+components that are unlikely to contain T2* or S0 signal.
+``mdl`` (conservative & fewest components), ``kic``,
+& ``aic`` (liberal & more components) use `MAPCA`_.
+Within the same general method, each uses a cost function to find a minimum
+where more components no longer model meaningful variance.
+For some datasets we see all methods fail and result in too few or too many components.
+There is no consistent number of components or % variance explained to define the correct number.
+The correct number of components will depend on the noise levels of the data.
+For example, smaller voxels will results in more thermal noise and less total variance explained.
+A dataset with more head motion artifacts will have more variance explained,
+since more structured signal is within the head motion artifacts.
+The clear failure cases are extreme. That is getting less than 1/5 the number of components
+compared to time points or having nearly has many components as time point.
+We are working on identifying why this happens and adding get better solutions.
+Our current guess is that most of the above methods assume data are
+independant and identically distributed (IID),
+and signal leakage from in-slice and multi-slice accelleration may violate this assumption.
+
+We have one option that is generally useful and is also a partial solution.
+``--ica_method robustica`` will run `robustica`_.
+This is a method that, for a given number of PCA components,
+will repeated run ICA and identify components that are stable across iterations.
+While running ICA multiple times will slow processing, as a general benefit,
+this means that the ICA results are less sensitive to the initialization parameters,
+computer hardware, and software versions.
+This will result in better stability and replicability of ICA results.
+Additionally, `robustica`_ almost always results in fewer components than initially prescripted,
+since there are fewer stable components across interations than the total number of components.
+This means, even if the initial PCA component estimate is a bit off,
+the number of resulting robust ICA components will represent stable information in the data.
+For a dataset where the PCA comoponent estimation methods are failing,
+one could use ``--tedpca`` with a fixed integer for a constant number of components,
+that is on the high end of typical for a study,
```

Comment:
```suggestion
one could use ``--tedpca`` in such a way that very few components are removed
(eg. using a fixed integer for a constant number of components
that is close to the number of volumes,
or a floating-point number just below 1.0),
```

Worth noting here that having looked at a few PCA variance plots with @BahmanTahayori, I think it should be possible to derive a PCA rank estimation algorithm that is specifically tailored to operate in conjunction with RobustICA, particularly in the use case where PCA-based denoising has been applied further upstream. It's not critical to the PR so don't want to delay based on such, but hopefully we can start a separate thread on it at some point.

Reply:
Your addition to this comment has caused issues on my data. Without running MP-PCA first, like you're doing, this results in each ICA iteration failing to converge and then the ICA iterations that do converge are sometimes weird. PIcking a number of components that is at the high end of typical for a study resulted in much more stable results (this might be one more thing to investigate more carefully when this PR is merged)

FWIW, I am also evaluating some PCA rank estimation options that can also be added in.

Reply:
Fair enough. This maybe instead needs to fall under the banner of advice for our recommended pipeline as a whole; ie. if replicating what we've done in our study, including upstream denoising, then we advise modulating the TEDANA PCA behaviour in this way. We'll have to wait to see whether such advice should appear in TEDANA documentation or elsewhere. But if RobustICA is being run on data without upstream denoising, and you've found that upping the PCA rank a bit but not all the way seems to help, then the text you have suits, and indeed hopefully we'll get more feedback over time.